bundles/alpha.model/src/alpha/model/util/ISLUtil.xtend

```diff
@@ -0,0 +1,114 @@
+package alpha.model.util
+
+import fr.irisa.cairn.jnimap.isl.ISLAff
+import fr.irisa.cairn.jnimap.isl.ISLBasicMap
+import fr.irisa.cairn.jnimap.isl.ISLBasicSet
+import fr.irisa.cairn.jnimap.isl.ISLConstraint
+import fr.irisa.cairn.jnimap.isl.ISLContext
+import fr.irisa.cairn.jnimap.isl.ISLDimType
+import fr.irisa.cairn.jnimap.isl.ISLMatrix
+import fr.irisa.cairn.jnimap.isl.ISLSet
+
+import static extension alpha.model.matrix.MatrixOperations.scalarMultiplication
+import static extension alpha.model.matrix.MatrixOperations.transpose
+import static extension alpha.model.util.DomainOperations.*
+
+class ISLUtil {
+	
+	/** Creates an ISLBasicSet from a string */
+	def static toISLBasicSet(String descriptor) {
+		ISLBasicSet.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Creates an ISLSet from a string */
+	def static toISLSet(String descriptor) {
+		ISLSet.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Creates an ISLBasicMap from a string */
+	def static toISLBasicMap(String descriptor) {
+		ISLBasicMap.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Creates an ISLBasicSet from a string */
+	def static toISLAff(String descriptor) {
+		ISLAff.buildFromString(ISLContext.instance, descriptor)
+	}
+	
+	/** Transposes an ISLMatrix */
+	def static transpose(ISLMatrix matrix) {
+		ISLMatrix.buildFromLongMatrix(matrix.toLongMatrix.transpose)
+	}
+	
+	/** Returns the integer point closest to the origin in set without parameter context */
+	def static long[] integerPointClosestToOrigin(ISLBasicSet set) {
+		set.copy.samplePoint.coordinates.subList(set.nbParams, set.nbParams + set.nbIndices)
+	}
+	
+	def static isTrivial(ISLBasicSet set) {
```

Comment:
To avoid duplicate implementations, let's implement `isTrivial(ISLBasicSet set)` by just calling `set.copy.toSet.isTrivial`.

Reply:
Done